jdbi/src/main/java/org/killbill/commons/jdbi/guice/DataSourceProvider.java

```diff
@@ -24,20 +24,22 @@
 import java.sql.SQLException;
 import java.util.concurrent.TimeUnit;
 
+import javax.inject.Provider;
 import javax.sql.DataSource;
 
 import org.killbill.commons.embeddeddb.EmbeddedDB;
 import org.killbill.commons.health.api.HealthCheckRegistry;
 import org.killbill.commons.jdbi.hikari.KillBillHealthChecker;
 import org.killbill.commons.jdbi.hikari.KillBillMetricsTrackerFactory;
 import org.killbill.commons.metrics.api.MetricRegistry;
+import org.killbill.commons.utils.annotation.VisibleForTesting;
 import org.skife.config.TimeSpan;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.annotations.VisibleForTesting;
+// FIXME-1615: Still using guice because metricRegistry and healthCheckRegistry need optional injection
+//   not sure what going to do with this. Probably leave this as is.
```

Comment:
Maybe something like this would work?

```
@Inject
@Nullable
public void setMetricsRegistry(final MetricRegistry metricRegistry)
```

or

```
@Inject
public void setMetricsRegistry(final Optional<MetricRegistry> metricRegistry)
```

If not, add the arguments to the `Inject` constructor. Users of `DataSourceProvider` could `bind` a `null` provider for these arguments.

Reply:
@pierre fixed. FYI the one that works is something like this (and this is what we have now):
```java
    @Inject
    public void setMetricsRegistry(@Nullable final MetricRegistry metricRegistry) {
        this.metricRegistry = metricRegistry;
    }
```